src/RA_AchievementOverlay.cpp

```diff
@@ -795,27 +795,30 @@ void AchievementOverlay::DrawAchievementExaminePage(HDC hDC, int nDX, int nDY, c
             TextOut(hDC, nDX + nRecentWinnersSubtitleX, nRecentWinnersSubtitleY, NativeStr(buffer).c_str(), strlen(buffer));
         }
 
-        for (unsigned int i = 0; i < g_AchExamine.NumRecentWinners(); ++i)
+        auto i{ 0U };
+        for (const auto& data : g_AchExamine)
         {
-            const AchievementExamine::RecentWinnerData& data = g_AchExamine.GetRecentWinner(i);
-
-            char buffer[256];
-            sprintf_s(buffer, 256, " %s ", data.User().c_str());
-
-            char buffer2[256];
-            sprintf_s(buffer2, 256, " %s ", data.WonAt().c_str());
+            {
+                auto sUser{ " "s };
+                sUser += data.User();
+                sUser += " ";
```

Comment:
Alright, using std::string explicitly instead of auto will work. The sequence of "+" is slow though, it's much faster to use += or append if all the arguments are strings.